orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/DependentPipelineStarter.groovy

```diff
@@ -101,28 +102,32 @@ class DependentPipelineStarter implements ApplicationContextAware {
     //keep the trigger as the preprocessor removes it.
     def expectedArtifacts = pipelineConfig.expectedArtifacts
 
-    for (ExecutionPreprocessor preprocessor : executionPreprocessors.findAll {
-      it.supports(pipelineConfig, ExecutionPreprocessor.Type.PIPELINE)
-    }) {
-      pipelineConfig = preprocessor.process(pipelineConfig)
-    }
-
     if (parentPipelineStageId != null) {
       pipelineConfig.receivedArtifacts = artifactResolver?.getArtifacts(parentPipeline.stageById(parentPipelineStageId))
     } else {
       pipelineConfig.receivedArtifacts = artifactResolver?.getAllArtifacts(parentPipeline)
     }
 
-    pipelineConfig.trigger = trigger
-    pipelineConfig.expectedArtifacts = expectedArtifacts
-
     def artifactError = null
     try {
       artifactResolver?.resolveArtifacts(pipelineConfig)
     } catch (Exception e) {
       artifactError = e
     }
 
+    for (ExecutionPreprocessor preprocessor : executionPreprocessors.findAll {
+      it.supports(pipelineConfig, ExecutionPreprocessor.Type.PIPELINE)
+    }) {
+      pipelineConfig = preprocessor.process(pipelineConfig)
+    }
+
+    if (pipelineConfig.errors != null) {
+      throw new ValidationException("Pipeline template is invalid", pipelineConfig.errors as List<Map<String, Object>>)
```

Comment:
Template parsing errors were silently being swallowed, so I added this to surface them to Deck in a proper way. Without this, the code will fail [here](https://github.com/spinnaker/orca/pull/2963/files#diff-1c0229177d2a2ff70e69b3e0ef8b54c9R181) with the error `Cannot get property 'id' on null object`.